server/model/pagination.js

```diff
@@ -8,27 +8,53 @@ export type Pagination = {|
   currentPage: number;
   nextPage?: ?number;
   prevPage?: ?number;
+  nextQueryString?: ?string;
+  prevQueryString?: ?string;
 |}
 
 export class PaginationFactory {
-  static fromList(l: List<any>, total: number, currentPage: number = 1): Pagination {
+  static fromList(l: List<any>, total: number, currentPage: number = 1, pageSize: number = 32, getParams: {} = {}): Pagination {
     const size = l.size;
-    const pageCount = Math.ceil(total / l.size);
+    const pageCount = Math.ceil(total / pageSize);
     const prevPage = pageCount > 1 && currentPage !== 1 ? currentPage - 1 : null;
     const nextPage = pageCount > 1 && currentPage !== pageCount ? currentPage + 1 : null;
+    const beginning = (pageSize * currentPage) - pageSize + 1;
     const range = {
-      beginning: (size * currentPage) - size + 1,
-      end: size * currentPage
+      beginning: beginning,
+      end: beginning + size - 1
     };
-
+    const nextQueryString = buildQueryString(nextPage, getParams);
```

Comment:
I have no idea – what's the SOP? I guess it would make our code cleaner. Maybe we should have a chat about it?